src/postamble.js

```diff
@@ -169,7 +169,7 @@ function run() {
     return;
   }
 
-  function doRun() {
+  async function doRun() {
```

Comment:
You'll want to do the same thing as `callMain` higher up in the file with `{{{ asyncIf(ASYNCIFY == 2) }}}`
and `{{{ awaitIf(ASYNCIFY == 2) }}}`.

Reply:
Thanks - awaitIf was the missing piece of the puzzle.